src/ocramius/util/Optional.php

```diff
@@ -188,6 +187,14 @@ public function filter(callable $predicate)
      * and if the result is non-null, return an {@code Optional} describing the
      * result.  Otherwise return an empty {@code Optional}.
      *
+     * @param callable(T):T $mapper a mapping function to apply to the value, if present
```

Comment:
Doesn't seem right: why is the mapper enforced to return a value of the same type of the input?

Reply:
You are right on this one, I wasn't thorough enough when checking the Java Docs regarding this function. I have changed the annotations to reflect that the mapper is of course allowed to return other types.